server/src/api/post.rs

```diff
@@ -578,20 +586,22 @@ impl Perform for Oper<EditPost> {
     let (iframely_title, iframely_description, iframely_html, pictrs_thumbnail) =
       fetch_iframely_and_pictrs_data(&self.client, data.url.to_owned()).await;
 
-    let edit_id = data.edit_id;
-    let read_post = blocking(pool, move |conn| Post::read(conn, edit_id)).await??;
+    // only modify some properties if they are a moderator
+    let post_stickied = Some(read_post.stickied);
+    let post_locked = Some(read_post.locked);
+    let post_removed = Some(read_post.removed);
 
     let post_form = PostForm {
       name: data.name.trim().to_owned(),
       url: data.url.to_owned(),
       body: data.body.to_owned(),
-      creator_id: data.creator_id.to_owned(),
-      community_id: data.community_id,
-      removed: data.removed.to_owned(),
+      creator_id: read_post.creator_id.to_owned(),
+      community_id: read_post.community_id,
+      removed: post_removed.to_owned(),
```

Comment:
I don't see how this is being done here. 

It should be a 

`let post_stickied = if moderators.contains(&user_id) { data.stickied } else { read_post...`